sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/PushNotifier.java

```diff
@@ -227,13 +275,38 @@ private static void setIcon(Context context, Intent pushIntent, Notification.Bui
     }
 
     private static int validateIcon(Context context, int iconResourceId) {
+        if (iconResourceId == 0) {
+            return iconResourceId;
+        }
         if (Build.VERSION.SDK_INT == Build.VERSION_CODES.O && context.getDrawable(iconResourceId) instanceof AdaptiveIconDrawable) {
             AppCenterLog.error(LOG_TAG, "Adaptive icons make Notification center crash (system process) on Android 8.0 (was fixed on Android 8.1), " +
                     "please update your icon to be non adaptive or please use another icon to push.");
             iconResourceId = 0;
         }
         return iconResourceId;
     }
+
+    private static int getResourceIdFromMetadata(Context context, String metadataName) {
+        Bundle metaData = null;
+        try {
+            metaData = context.getPackageManager().getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA).metaData;
+        } catch (PackageManager.NameNotFoundException e) {
+            AppCenterLog.error(LOG_TAG, "Package name not found.", e);
+        }
+        if (metaData != null) {
+            return metaData.getInt(metadataName);
+        }
+        return 0;
+    }
+
+    private static int getColor(Context context, int colorResourceId) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            return context.getColor(colorResourceId);
+        } else {
+            //noinspection deprecation
```

Comment:
move to method annotation otherwise we'll still get a warning in gradle build for that one.